src/main/java/com/company/WordCounter.java

```diff
@@ -0,0 +1,82 @@
+package com.company;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class WordCounter {
+
+    private String name;
+    private String flag;
+
+    WordCounter(String name, String flag) {
+        this.name = name;
+        this.flag = flag;
+        function(name, flag);
+    }
+
+    public void function(String name, String flag) {
+
+        try {
+            Scanner read = new Scanner(new File(name));
+            int numberOfLines = 0, numberOfWords = 0, numberOfCharacters = 0;
+            while (read.hasNextLine()) {
+                String line = read.nextLine();
+                numberOfLines++;
+                numberOfWords += counterWord(line);
+                numberOfCharacters += counterCharacter(line);
+            }
+
+            System.out.print(name + " : ");
+            switch (flag) {
```

Comment:
let's validate flag before we read file. If wrong one is passed, there is no point in reading the file.